openmls/src/framing/mls_auth_content.rs

```diff
@@ -170,21 +170,20 @@ impl AuthenticatedContent {
     }
 
     /// This constructor builds an `PublicMessage` containing an External Proposal.
-    /// The sender is [Sender::NewMemberProposal].
-    // TODO #151/#106: We don't support preconfigured senders yet
     pub(crate) fn new_external_proposal(
         proposal: Proposal,
         group_id: GroupId,
         epoch: GroupEpoch,
         signer: &impl Signer,
+        sender: Sender,
```

Comment:
I'd rather continue the separation here as well and keep the explicit sender type and introduce a `new_join_proposal` instead.